src/kernels/kernelRefreshIndicator.node.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { notebooks } from 'vscode';
+import { IExtensionSyncActivationService } from '../platform/activation/types';
+import { IPythonExtensionChecker } from '../platform/api/types';
+import { InteractiveWindowView, JupyterNotebookView } from '../platform/common/constants';
+import { disposeAllDisposables } from '../platform/common/helpers';
+import { IDisposable, IDisposableRegistry } from '../platform/common/types';
+import { IInterpreterService } from '../platform/interpreter/contracts';
+import { IKernelFinder } from './types';
+
+/**
+ * Ensures we refresh the list of Python environments upon opening a Notebook.
+ */
+@injectable()
+export class KernelRefreshIndicator implements IExtensionSyncActivationService {
+    private refreshedOnceBefore?: boolean;
```

Comment:
nit: Better to init to false so narrow the type, V8 maybe be able to optimize a `false | true` where it could not a `false | true | undefined`

Reply:
look, i thought it was the opposite, i thought allocating space for a default value of false unnecessarily was less optimal

Reply:
I think you're right wrt allocating space, it will likely deopt something in the class as its properties change after the ctor has run. That's getting a bit too unnecessarily deep though, the main reason I commented was for maintainability as I would expect a `?: boolean` to have 3 states